charms/slurmd/src/charm.py

```diff
@@ -174,22 +172,24 @@ def _on_slurmctld_available(self, event: SlurmctldAvailableEvent) -> None:
         self._stored.slurmctld_available = True
 
         # Restart munged and slurmd after we write the event data to their respective locations.
-        if self._slurmd_manager.restart_munged():
-            logger.debug("## Munge restarted successfully")
-        else:
-            logger.error("## Unable to restart munge")
-
-        slurmd.restart()
+        try:
+            logger.debug("restarted munge successfully")
+            self._slurmd.munge.service.restart()
```

Comment:
Switched order on these lines? Looks like the debug message should be after the restart.

Reply:
Fixed in https://github.com/charmed-hpc/slurm-charms/pull/35/commits/c751b0a5e4b24585aa05d37715e3c58526fb2303